src/otx/algo/hooks/recording_forward_hook.py

```diff
@@ -411,48 +390,45 @@ def create_and_register_hook(cls, num_classes: int) -> BaseRecordingForwardHook:
 
     def func(
         self,
-        preds: list[InstanceSegBatchPredEntity | InstanceSegBatchPredEntityWithXAI],
+        predictions: list[InstanceSegBatchPredEntity],
         _: int = -1,
     ) -> list[np.array]:
         """Generate saliency maps from predicted masks by averaging and normalizing them per-class.
 
         Args:
-            preds (List[InstanceSegBatchPredEntity]): Predictions of Instance Segmentation model.
+            preds (InstanceSegBatchPredEntity | dict): Predictions of Instance Segmentation model.
 
         Returns:
             list[np.array]: Class-wise Saliency Maps. One saliency map per each class - [batch, class_id, H, W]
         """
         # TODO(gzalessk): Add unit tests # noqa: TD003
-        batch_size = len(preds)
+        batch_size = len(predictions)
```

Comment:
If we use .append() in the loop over `predicitons`, we can avoid explicit list allocation and simplify the loop as a bonus.

Reply:
Good point